instrumentation/cassandra/cassandra-3.0/javaagent/src/test/java/CassandraClientTest.java

```diff
@@ -0,0 +1,422 @@
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+
+public class CassandraClientTest {
+
+  private static final Logger logger = LoggerFactory.getLogger(CassandraClientTest.class);
+
+  private static final Executor executor = Executors.newCachedThreadPool();
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @SuppressWarnings("rawtypes")
+  private static GenericContainer cassandra;
+
+  private static int cassandraPort;
+  private static Cluster cluster;
+
+  @BeforeAll
+  static void beforeAll() {
+    cassandra =
+        new GenericContainer<>("cassandra:3")
+            .withEnv("JVM_OPTS", "-Xmx128m -Xms128m")
+            .withExposedPorts(9042)
+            .withLogConsumer(new Slf4jLogConsumer(logger))
+            .withStartupTimeout(Duration.ofMinutes(2));
+    cassandra.start();
+
+    cassandraPort = cassandra.getMappedPort(9042);
+    cluster =
+        Cluster.builder()
+            .addContactPointsWithPorts(new InetSocketAddress("localhost", cassandraPort))
+            .build();
+  }
+
+  @AfterAll
+  static void afterAll() {
+    cluster.close();
+    cassandra.stop();
+  }
+
+  @ParameterizedTest(name = "{index}: {0}")
+  @MethodSource("provideSyncParameters")
+  void syncTest(Parameter parameter) {
+    Session session = cluster.connect(parameter.keyspace);
+
+    session.execute(parameter.statement);
+
+    if (parameter.keyspace != null) {
+      testing.waitAndAssertTraces(
+          trace ->
+              trace.hasSpansSatisfyingExactly(
+                  span ->
+                      span.hasName("DB Query")
+                          .hasKind(SpanKind.CLIENT)
+                          .hasNoParent()
+                          .hasAttributesSatisfying(
+                              attributes ->
+                                  assertThat(attributes)
+                                      .containsEntry(NET_SOCK_PEER_ADDR, "127.0.0.1")
+                                      .containsEntry(NET_SOCK_PEER_NAME, "localhost")
+                                      .containsEntry(NET_SOCK_PEER_PORT, cassandraPort)
+                                      .containsEntry(DB_SYSTEM, "cassandra")
+                                      .containsEntry(DB_STATEMENT, "USE " + parameter.keyspace))
```

Comment:
can you change the assertions with fixed number of attributes to use `hasAttributesSatisfyingExactly`?

Reply:
```
.hasAttributesSatisfyingExactly(
      equalTo(DB_CASSANDRA_TABLE, null)
)
```
It works, in this case dividing is not necessary

Reply:
@trask Could you check my changes, please?